substratest/assets.py

```diff
@@ -194,13 +194,15 @@ class DataSample(_Asset):
 class ObjectiveDataset(_InternalStruct):
     dataset_key: str
     data_sample_keys: typing.List[str]
+    metadata: typing.Dict
 
 
 class Dataset(_Asset):
     key: str
     name: str
     owner: str
     objective_key: str
+    metadata: typing.Dict
```

Comment:
Since you specified in the backend that metadata is a dict of strings, you could also specify it here with the syntax typing.Dict[str, str] (keys are str, values are str)

Reply:
This would also apply to other assets.

Reply:
Yes, indeed!